src/model.cpp

```diff
@@ -120,8 +137,8 @@ bool Model::addUnits(const UnitsPtr &units)
         return false;
     }
 
-    // Prevent adding multiple times to list.
-    if (hasUnits(units)) {
+    // Prevent adding the same units.
+    if (std::find(mPimpl->mUnits.begin(), mPimpl->mUnits.end(), units) != mPimpl->mUnits.end()) {
         return false;
```

Comment:
Also here with the equality thing.  Why are we preventing it?

Reply:
Because this is the behaviour we have already decided upon.

Reply:
Well, no.  Everything in this PR is new behaviour, and the prevention of duplicates definitely belongs in the old behaviour set.  As far as I'm aware, we haven't ever discussed this in the new framework?  I find it very odd that you can delete, move, edit items based only on their content, but you can't add them that way?

Reply:
The old behaviour is to stop you adding the same pointer to the list.  The change here maintains that behaviour.

Reply:
I get it, but it doesn't make sense in the new context of thinking about items.  Why should the user have to care how we store the information?  Why can't you have two items the same on the list since their contents is now all we care about?